src/features/device-state/routes/state-patch-v2.ts

```diff
@@ -192,16 +192,25 @@ export const statePatchV2: RequestHandler = async (req, res) => {
 						// If we're updating anyway then ensure the metrics data is included
 						deviceBody = { ...deviceBody, ...metricsBody };
 						if (deviceBody.cpu_id != null) {
-							deviceBody.cpu_id = deviceBody.cpu_id.toLowerCase();
+							if (/[^\x20-\x7E]/.test(deviceBody.cpu_id)) {
+								// if the CPU id is not in the character range of 0x20 (SPACE) to 0x7e (~) we drop the CPU ID
+								// this cpu id wouldn't be rendered anyway
+								delete deviceBody.cpu_id;
+							} else {
+								deviceBody.cpu_id = deviceBody.cpu_id.toLowerCase();
+							}
+						}
```

Comment:
It would make more sense to move this up next to
```ts
			if (local.name != null) {
				deviceBody.device_name = local.name;
			}
```
so that the `Object.keys(deviceBody).length > 0` check will be handled in the standard way

Reply:
Oh I see, thanks